### ISSUE TITLE
controller build failure

### ISSUE THREAD

**Issue by zimbatm** [remote-tabs/remote-tabs#9]
I am getting the following error with snack, any ideas @nmattia ?
```
$ nix-build nix/release.nix -A rt-controller
these derivations will be built:
  /nix/store/15v7zmc26z0k32p9a2xgw76j6r4x6vbg-Main.drv
  /nix/store/13bfmdycnyzqbx8hdh2fnycnq30smmrx-linker-builder.drv
  /nix/store/mai1sqaqrzkrl5vi0xw5snps5i177dy3-linker.drv
building '/nix/store/15v7zmc26z0k32p9a2xgw76j6r4x6vbg-Main.drv'...
unpacking sources
unpacking source archive /nix/store/fjnp4y41pp4dwl0b5rpmkxrbs4cdm7ry-extra-files
source root is extra-files
/nix/store/15kgcm8hnd99p7plqzx7p4lcr2jni4df-set-source-date-epoch-to-latest.sh: line 13: [: : integer expression expected
building
Building module Main
Creating dependencies symtree for module Main
Creating module symlink for module Main
Compiling module Main

Main.hs:97:27: error:
    * Exception when trying to run compile-time code:
        ./pages/displays.html: openFile: does not exist (No such file or directory)
      Code: compileHtmlFile "./pages/displays.html"
    * In the untyped splice: $(compileHtmlFile "./pages/displays.html")
   |
97 |     in Blaze.renderMarkup $(compileHtmlFile "./pages/displays.html")
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
builder for '/nix/store/15v7zmc26z0k32p9a2xgw76j6r4x6vbg-Main.drv' failed with exit code 1
cannot build derivation '/nix/store/13bfmdycnyzqbx8hdh2fnycnq30smmrx-linker-builder.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/mai1sqaqrzkrl5vi0xw5snps5i177dy3-linker.drv': 1 dependencies couldn't be built
error: build of '/nix/store/mai1sqaqrzkrl5vi0xw5snps5i177dy3-linker.drv' failed


```

**Comment by zimbatm** [comment]
when patching snack:
```
diff --git a/snack/default.nix b/snack/default.nix
index 31d47d1..a4c08c0 100644
--- a/snack/default.nix
+++ b/snack/default.nix
@@ -133,6 +133,10 @@ let
         [ "unpackPhase" "buildPhase" ];
       buildPhase =
         ''
+          echo "========================================"
+          echo PWD=$PWD
+          ls -la
+          echo "========================================"
           echo "Building module ${mod.moduleName}"
           mkdir -p $out
           echo "Creating dependencies symtree for module ${mod.moduleName}"
```

```
$ snack build
snack nix: snack.nix
command: build
these derivations will be built:
  /nix/store/mcingkx5a0s38v5axz3f35pjqqhym878-Main.drv
  /nix/store/n6s1ipv7z59wpmp4xgsx52l55k1kxls9-linker-builder.drv
  /nix/store/803kaz1vaqsr4gzaww9n9jmca0nsz5p0-linker.drv
building '/nix/store/mcingkx5a0s38v5axz3f35pjqqhym878-Main.drv'...
unpacking sources
unpacking source archive /nix/store/fjnp4y41pp4dwl0b5rpmkxrbs4cdm7ry-extra-files
source root is extra-files
/nix/store/15kgcm8hnd99p7plqzx7p4lcr2jni4df-set-source-date-epoch-to-latest.sh: line 13: [: : integer expression expected
building
========================================
PWD=/build/extra-files
total 16
drwxr-xr-x 2 nixbld nixbld 4096 Jan  1  1970 .
drwx------ 3 nixbld nixbld 4096 May 25 15:04 ..
lrwxrwxrwx 1 nixbld nixbld   63 Jan  1  1970 displays.html -> /nix/store/kmvrlii5xas6b999qbw63ac4hys7apnd-pages/displays.html
lrwxrwxrwx 1 nixbld nixbld   60 Jan  1  1970 index.html -> /nix/store/kmvrlii5xas6b999qbw63ac4hys7apnd-pages/index.html
========================================
Building module Main
Creating dependencies symtree for module Main
Creating module symlink for module Main
Compiling module Main

Main.hs:97:27: error:
    * Exception when trying to run compile-time code:
        ./pages/displays.html: openFile: does not exist (No such file or directory)
      Code: compileHtmlFile "./pages/displays.html"
    * In the untyped splice: $(compileHtmlFile "./pages/displays.html")
   |
97 |     in Blaze.renderMarkup $(compileHtmlFile "./pages/displays.html")
   |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
builder for '/nix/store/mcingkx5a0s38v5axz3f35pjqqhym878-Main.drv' failed with exit code 1
cannot build derivation '/nix/store/n6s1ipv7z59wpmp4xgsx52l55k1kxls9-linker-builder.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/803kaz1vaqsr4gzaww9n9jmca0nsz5p0-linker.drv': 1 dependencies couldn't be built
error: build of '/nix/store/803kaz1vaqsr4gzaww9n9jmca0nsz5p0-linker.drv' failed
```

**Comment by zimbatm** [comment]
```
diff --git a/snack/default.nix b/snack/default.nix
index a4c08c0..61d8e33 100644
--- a/snack/default.nix
+++ b/snack/default.nix
@@ -127,7 +127,7 @@ let
     { name = objectName;
       src = pkgs.symlinkJoin
         { name = "extra-files";
-          paths = [ extraFiles ] ++ mod.moduleDirectories;
+          paths = pkgs.lib.traceValSeq ([ extraFiles ] ++ mod.moduleDirectories);
         };
       phases =
         [ "unpackPhase" "buildPhase" ];

```

gives me this:
```
trace: [ "/nix/store/7286xhidvg8faxwn2m7v9mjsn0pbaifm-src" /home/zimbatm/go/src/github.com/remote-tabs/remote-tabs/controller/pages ]
```

**Comment by nmattia** [comment]
Fixed in #10
